app/models/quota.rb

```diff
@@ -0,0 +1,112 @@
+# This describes disk quota utilization for a given user and volume
+class Quota
+  # Type of quota this object describes
+  # @return [Symbol] type of quota
+  attr_reader :type
+
+  # Path to volume
+  # @return [Pathname] path to volume
+  attr_reader :path
+
+  # User utilizing disk resources under this volume
+  # @return [String] user name
+  attr_reader :user
+
+  # Disk space used by user for this volume
+  # @return [Integer] blocks used by user
+  attr_reader :block_usage
+
+  # Disk space used by all users this quota encompasses for this volume
+  # @return [Integer] total blocks used
+  attr_reader :total_block_usage
+
+  # Disk space limit set for this volume
+  # @return [Integer] block limit
+  attr_reader :block_limit
+
+  # Number of files owned by user for this volume
+  # @return [Integer] files owned by user
+  attr_reader :file_usage
+
+  # Number of files owned by all users this quota encompasses for this volume
+  # @return [Integer] total files owned
+  attr_reader :total_file_usage
+
+  # Number of files limit set for this volume
+  # @return [Integer] file limit
+  attr_reader :file_limit
+
+  # When this quota report was generated by the file system
+  # @return [Time] when quota generated
+  attr_reader :updated_at
+
+  class << self
+    # Get quota objects for all users in JSON file(s)
+    def all(opts = {})
+      find(nil, opts)
+    end
+
+    # Get quota objects only for requested user in JSON file(s)
+    def find(user, opts = {})
+      user  = user && user.to_s
+      opts  = opts.to_h.compact.symbolize_keys
+      paths = opts.fetch(:paths, Configuration.quota_paths)
+      paths = Array.wrap(paths).map { |p| Pathname.new p }
+
+      quotas = []
+      paths.each do |path|
+        json = Oj.strict_load(path.read)
```

Comment:
In your tests, `Oj` was on avg 29ms vs regular `JSON` which was avg 47ms. 19ms 
difference. Not sure that this is worth adding another third party dependency.

Though for non-core OOD apps, both this dependency and https://github.com/ohler55/ox look interesting.